### PR TITLE
[Backport 2.x] Moving concurrent-search out of sandbox plugin to core behind feature…

### PR DESCRIPTION
**Pull request by kotwanikunal**
… flag (#7203)

Signed-off-by: Sorabh Hamirwasia <sohami.apache@gmail.com>
(cherry picked from commit fde8fa47a14b4dbb17a670604ab0ff357eeb5b95)

### Description
- Backports #7203 to `2.x`
- Fixes changelog diff b/w main and 2.x

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,4 +34,5 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### Security
 
-
+[Unreleased 3.0]: https://github.com/opensearch-project/OpenSearch/compare/2.x...HEAD
```
Comment (reta) [inline CHANGELOG.md:37]: Should not be there

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,9 +5,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased 2.x]
 ### Added
-- [Extensions] Moving Extensions APIs to support cross versions via protobuf. ([#6960](https://github.com/opensearch-project/OpenSearch/pull/6960))
+- [Extensions] Moving Extensions APIs to support cross versions via protobuf. ([#7402](https://github.com/opensearch-project/OpenSearch/issues/7402))
```
Comment (reta) [inline CHANGELOG.md:8]: The changelog looks a bit messed up

Comment (kotwanikunal) [reply CHANGELOG.md:8]: Updated this as per the merge on #7402
Updated the main changelog on #7409

File: CHANGELOG.md (hunk 1 above)
Comment (kotwanikunal) [reply CHANGELOG.md:37]: I will remove it with a new PR.

File: CHANGELOG.md (hunk 2 above)
Comment (saratvemulapalli) [reply CHANGELOG.md:8]: My bad @reta. I've started merging in more PRs but it didn't make sense for readers as all of them are just moving these APIs to support cross versions. So I've just moved it to meta issue and updated main and 2.x.

**Review by tlfeng — APPROVED** [review]

**Pull request merged**